src/Migrate/custom/New-AzMigrateServerReplication.ps1

```diff
@@ -464,6 +514,251 @@ public static int hashForArtifact(String artifact)
+            {
+                throw "InvalidTagNameCharacters : The tag names '$($IllegalCharKey -join ', ')' have reserved characters '<,>,%,&,\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'."
+            }
+
+            if ($ExceededLengthKey.Count -gt 0)
+            {
+                throw "InvalidTagName : Tag value too large. Following tag value '$($ExceededLengthKey -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '512' characters."
+            }
+
+            if ($ExceededLengthValue.Count -gt 0)
+            {
+                throw "InvalidTagValueLength : Tag value too large. Following tag value '$($ExceededLengthValue -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '256' characters."
+            }
+
+            $ProviderSpecificDetails.SeedDiskTag = $Tag
+            $ProviderSpecificDetails.TargetDiskTag = $Tag
+            $ProviderSpecificDetails.TargetNicTag = $Tag
+            $ProviderSpecificDetails.TargetVmTag = $Tag
+        }
+
+        if ($HasVMTag) {
+            $IllegalCharKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthValue = New-Object Collections.Generic.List[String]
+
+            if ($VMTag.Count -gt 50)
+            {
+                throw "InvalidTags : Too many tags specified. Requested tag count - '$($VMTag.Count)'. Maximum number of tags allowed - '50'."
+            }
+
+            foreach ($x in $VMTag.Keys)
+            {
+                if ($x.length -gt 512)
+                {
+                    $ExceededLengthKey.add($x)
+                }
+
+                if ($x -match "[<>%&\?/.]")
+                {
+                    $IllegalCharKey.add($x)
+                }
+
+                if ($($VMTag.Item($x)).length -gt 256)
+                {
+                    $ExceededLengthValue.add($($VMTag.Item($x)))
+                }
+            }
+
+            if ($IllegalCharKey.Count -gt 0)
+            {
+                throw "InvalidTagNameCharacters : The tag names '$($IllegalCharKey -join ', ')' have reserved characters '<,>,%,&,\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'."
+            }
+
+            if ($ExceededLengthKey.Count -gt 0)
+            {
+                throw "InvalidTagName : Tag value too large. Following tag value '$($ExceededLengthKey -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '512' characters."
+            }
+
+            if ($ExceededLengthValue.Count -gt 0)
+            {
+                throw "InvalidTagValueLength : Tag value too large. Following tag value '$($ExceededLengthValue -join ', ')' exceeded the maximum length. Maximum allowed length for tag value - '256' characters."
```

Comment:
Check with Rahul Gupta/Deepika.

Reply:
Error message is shared by Deepika and it is same as what compute throws.